src/sec.rs

```diff
@@ -290,24 +290,17 @@ impl SecClient {
     }
 }
 
-/*
- * TODO-cleanup Is this necessary?  Correct?
- * TODO-correctness Sending messages to the SEC can fail if it panicked.  What
- * do we want to do here?
- */
 impl Drop for SecClient {
     fn drop(&mut self) {
         if !self.shutdown {
             /*
-             * If we get here, there must be no outstanding requests on this
-             * channel.  Thus, there must be buffer space, and try_send() ought
-             * not to fail for running out of space.  It also ought not to fail
-             * because the other side is closed either.  See shutdown() for
-             * details.
+             * If we get here, there should be no outstanding requests on this
+             * channel, in which case there must be buffer space and try_send()
+             * ought not to fail for running out of space.  It may fail if the
+             * other side is closed, but that should only happen if the SEC task
+             * panicked.
```

Comment:
And, fortunately, for non-test code where "panic = abort", this will bring down the whole process - so in that case, we won't need to worry about it silently getting away from us.